nixos/modules/services/continuous-integration/buildkite-agents.nix

```diff
@@ -68,19 +79,23 @@ in
 
       name = mkOption {
         type = types.str;
-        default = "%hostname-%n";
+        default = "%hostname-${name}-%n";
         description = ''
-          The name of the agent.
+          The name of the agent as seen in the buildkite dashboard.
         '';
       };
 
-      meta-data = mkOption {
-        type = types.str;
-        default = "";
-        example = "queue=default,docker=true,ruby2=true";
+      tags = mkOption {
+        type = let
+          commasToAttrs = commas: builtins.foldl'
+            (prev: cur: let pair = builtins.split "=" cur; in
+                 prev // {"${lib.head pair}" = lib.last pair; })
+            {} (lib.remove [] (builtins.split "," commas)); in
+          types.coercedTo types.str commasToAttrs (types.attrsOf types.str);
```

Comment:
This is rather ugly just to support legacy configuration. I suggest putting this conversion into a `mkChangedOptionModule` instead so this type can be just `attrsOf str`.

Reply:
Given we're changing a lot of things in the module anyways, and break backwards compat, we can probably get rid of the old syntax too, WDYT?

Reply:
Sounds good to me, a `mkRemovedOptionModule` it would be then

Reply:
The whole module was changed significantly to allow multiple buildkite agents to be deployed -

I'm not that much a fan of tracking every removed suboption in the imports too - I added a

```
imports = [
    (mkRemovedOptionModule [ "services" "buildkite-agent"] "services.buildkite-agent has been moved to an attribute set at services.buildkite-agents")
  ];
```

do you think that's sufficient?

Reply:
Yeah that's good